src/main/java/org/folio/circulation/domain/notice/schedule/RequestScheduledNoticeHandler.java

```diff
@@ -0,0 +1,118 @@
+package org.folio.circulation.domain.notice.schedule;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.support.Result.succeeded;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.Request;
+import org.folio.circulation.domain.RequestAndRelatedRecords;
+import org.folio.circulation.domain.RequestRepository;
+import org.folio.circulation.domain.notice.NoticeTiming;
+import org.folio.circulation.domain.notice.PatronNoticeService;
+import org.folio.circulation.domain.notice.TemplateContextUtil;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import io.vertx.core.json.JsonObject;
+
+public class RequestScheduledNoticeHandler {
+
+  public static RequestScheduledNoticeHandler using(Clients clients) {
+    return new RequestScheduledNoticeHandler(
+      RequestRepository.using(clients, true),
+      PatronNoticeService.using(clients),
+      ScheduledNoticesRepository.using(clients));
+  }
+
+  private RequestRepository requestRepository;
+  private PatronNoticeService patronNoticeService;
+  private ScheduledNoticesRepository scheduledNoticesRepository;
+
+  private RequestScheduledNoticeHandler(RequestRepository requestRepository,
+                                        PatronNoticeService patronNoticeService,
+                                        ScheduledNoticesRepository scheduledNoticesRepository) {
+
+    this.requestRepository = requestRepository;
+    this.patronNoticeService = patronNoticeService;
+    this.scheduledNoticesRepository = scheduledNoticesRepository;
+  }
+
+  public CompletableFuture<Result<Collection<ScheduledNotice>>> handleNotices(Collection<ScheduledNotice> scheduledNotices) {
+    CompletableFuture<Result<ScheduledNotice>> future = completedFuture(succeeded(null));
+    for (ScheduledNotice scheduledNotice : scheduledNotices) {
+      future = future.thenCompose(r -> r.after(v -> handleRequestNotice(scheduledNotice)));
```

Comment:
It is intended that these are done in sequence one after the other?

Reply:
Right. I used this approach as it was implemented in the same way for due date notices. From what I see it can be done in parallel. Probably it can give better performance.